### PR TITLE
Payment registry

### LINKED ISSUE #1744: refactor: Improvements to Cash Payments Registry
This issue is to track improvements to the Cash Payments Registry to be tackled while the team is working all registries.

Parent Issue #1737.

-----

The follow changes should be made to the Cash Payments:
 - [x] Make a "Menu" dropdown with the following options:
   1. Column Configuration
   2. Saving the Grid Configuration
   3. Resetting the Grid Configuration
   4. Download View as CSV
   5. Download View as PDF
 - [x] Implement the Grid State work on the grid (See #1738)
 - [x] Make sure Search Form is up to standards with `bhClear` on each optional input (see #1702)
 - [x] Remove the `bhRendererDropdown`
 - [ ] Implement downloading as CSV using the `ui-grid` exporter (See https://github.com/IMA-WorldHealth/bhima-2.X/pull/1554)

### PR DESCRIPTION
**Pull request by DedrickEnc**
This PR addresses the issues #1744 
Now the cash registry is following the new standard
closes #1744

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by DedrickEnc** [comment]
~~I habe some missed test cases, I should complete them first before this PR be reviewed~~

File: client/src/i18n/en/cash_registry.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "CASH_REGISTRY" : {
+        "TITLE" : "Payment registry",
+        "RECEIPT_CURRENCY" : "receipt Currency"
```
Comment (sfount) [inline client/src/i18n/en/cash_registry.json]: I don't think this key for RECEIPT_CURRENCY is used.

File: client/src/modules/cash/payments/registry.js [hunk 2]
```diff
@@ -170,5 +189,11 @@ function CashPaymentRegistryController(
     vm.latestViewFilters = Cash.filters.formatView();
   }
 
+  // This function opens a modal through column service to let the user toggle
```
Comment (sfount) [inline client/src/modules/cash/payments/registry.js:192]: Useful comments 👍

**Review by sfount — COMMENTED** [review]
@DedrickEnc Code LGTM, great work 👍 . 

I have downloaded and tested this branch and it seems to work well. One thing that I've noticed is that the 'Download as PDF' and 'Download as CSV' options do not factor in the current filters - the files are created with all data no matter what the current parameters. 

It would be best if these downloaded files respected the current filters (an example of this is the Invoice registry).

**Comment by DedrickEnc** [comment]
@sfount
This PR is fixed

**Comment by sfount** [comment]
@DedrickEnc I have tested the report filtering. Everything looks good to me!

**Pull request merged**